README.md

```diff
@@ -29,99 +31,99 @@ Tested in Node 4+.
-The function [`search()`](#searchsearchpath) searches for a configuration object and returns a Promise,
-which resolves with an object containing the information you're looking for.
-
-You can also pass option `sync: true` to search or load the config synchronously, returning the config itself.
-
-So let's say `var yourModuleName = 'goldengrahams'` — here's how cosmiconfig will work:
-
-- Starting from `process.cwd()` (or some other directory defined by the `searchPath` argument to `search()`), it looks for configuration objects in three places, in this order:
-  1. A `goldengrahams` property in a `package.json` file (or some other property defined by `options.packageProp`);
-  2. A `.goldengrahamsrc` file with JSON or YAML syntax (or some other filename defined by `options.rc`);
-  3. A `goldengrahams.config.js` JS file exporting the object (or some other filename defined by `options.js`).
-- If none of those searches reveal a configuration object, it moves up one directory level and tries again. So the search continues in `./`, `../`, `../../`, `../../../`, etc., checking those three locations in each directory.
-- It continues searching until it arrives at your home directory (or some other directory defined by `options.stopDir`).
-- If at any point a parseable configuration is found, the `search()` Promise resolves with its result object.
-- If no configuration object is found, the `search()` Promise resolves with `null`.
-- If a configuration object is found *but is malformed* (causing a parsing error), the `search()` Promise rejects and shares that error (so you should `.catch()` it).
-
-If you know where your configuration file is located you should instead use [`load()`](#loadconfigpath) which takes an optional string parameter `configPath`
-that defaults to [`options.configPath`](#configpath).  Cosmiconfig will read that file and try parsing it as JSON, YAML, or JS.
+## API
 
-## Caching
+### Result
 
-As of v2, cosmiconfig uses a few caches to reduce the need for repetitious reading of the filesystem. Every new cosmiconfig instance (created with `cosmiconfig()`) has its own caches.
+The result object you get from `search` or `load` has the following properties:
 
-To avoid or work around caching, you can
-- create separate instances of cosmiconfig, or
-- set `cache: false` in your options.
-- use the cache clearing methods documented below.
+- **config:** The parsed configuration object. `undefined` if the file is empty.
+- **filepath:** The path to the configuration file that was found.
+- **isEmpty:** `true` if the configuration file is empty.
 
-## API
+### Create an explorer
 
-### `var explorer = cosmiconfig(moduleName[, options])`
+```js
+const explorer = cosmiconfig(moduleName[, cosmiconfigOptions])
+```
 
-Creates a cosmiconfig instance (i.e. explorer) configured according to the arguments, and initializes its caches.
+Creates a cosmiconfig instance ("explorer") configured according to the arguments, and initializes its caches.
```

Comment:
Not sure whether you and @sudosuhas already discussed this, but do you have thoughts on normalizing the terminology surrounding **cosmiconfig instance** and **explorer**?  Another term used right under '## Usage' is **cosmiconfig explorer**.

This might be solved by an api change with the method 'createExplorer' being a property of cosmiconfig.  Anything besides that leaves room for ambiguity - which is not a big deal.  Just wanted to see what you guys thought.

Reply:
I don't think we need to make any API change; but if we wanted to change the docs we could do this:

```js
// Change this:
const cosmiconfig = require('cosmiconfig');
// To this:
const createExplorer = require('cosmiconfig');
```

Reply:
haha true